lambda_layers/core_layer/python/core_layer/handler/url_threatcheck.py

```diff
@@ -0,0 +1,66 @@
+import os
+import requests
+from ml_service.SearchFactChecks import get_secret
+
+API_URL_GOOGLE = "https://safebrowsing.googleapis.com/v4/threatMatches:find"
+THREAT_PREFIX_GOOGLE = "GOOGLE:"
+
+def threatcheck(url):
+    """ Checks the URL for various threats
+     Parameters
+    ----------
+     url:
+        string
+    Returns
+    ------
+        threat or None
+    """
+    result = get_googleapi_safebrowsing_threat(url)
+    return result
+
+def get_googleapi_safebrowsing_threat(url):
+    """ Calls Google API for checking unsafe urls
+    Parameters
+    ----------
+    url:
+        string, required
+    Returns
+    ------
+        for threat urls:
+            'GOOGLE:' followed by threat type (e.g. GOOGLE:MALWARE).
+        in case of failing API call:
+            GOOGLE:N/A
+        for no threat urls:
+            None
+    """
+    body = {
+        "client": {
+            "clientId": "codetect",
+            "clientVersion": "1.5.2"
+        },
+        "threatInfo": {
+            "threatTypes": ["MALWARE", "SOCIAL_ENGINEERING", "THREAT_TYPE_UNSPECIFIED",
+                            "UNWANTED_SOFTWARE", "POTENTIALLY_HARMFUL_APPLICATION"],
+            "platformTypes": ["ANY_PLATFORM"],
+            "threatEntryTypes": ["URL"],
+            "threatEntries": [{"url": url}]
+        }
+    }
+
+    ## first check if google api key is stored in environment parameter
+    key = os.environ["UNITTEST_GOOGLE_API_KEY"]
```

Comment:
Why do you safe this in an environment variable? The get secret call should also work, if your aws credentials are configured locally, shouldn't it?

Reply:
Yes, but not if the boto3 client is mocked (in order to mock the stepfunction and mail functionality), like in the testcase **test_submit_item2.py**

Reply:
Alright, I did not know that. Thanks!